node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -273,22 +274,27 @@ public void start(long stateConvergeInterval) {
         loopThread.start();
     }
 
-    @Override
-    public void deconstruct() {
+    public void stop() {
+        specVerifierScheduler.shutdown();
         if (!terminated.compareAndSet(false, true)) {
             throw new RuntimeException("Can not re-stop a node agent.");
         }
         log.log(LogLevel.INFO, objectToString() + ": Deconstruct called");
+
+        // First we need to stop NodeAdminStateUpdater thread to make sure no new NodeAgents are spawned
         signalWorkToBeDone();
-        try {
-            loopThread.join(10000);
-            if (loopThread.isAlive()) {
-                log.log(LogLevel.ERROR, "Could not stop tick thread");
+
+        do {
+            try {
+                loopThread.join(0);
+                specVerifierScheduler.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
```

Comment:
Would it be possible to encapsulate these executor services in a class to hide away the stopping of them? They're otherwise polluting the stop functions of several classes.